openapiv2/core/service.swagger.yaml

```diff
@@ -928,6 +926,54 @@ paths:
           pattern: '[^/]+/[^/]+'
       tags:
         - Credit
+  /v1beta/{owner_name}/credit/purchases:
+    get:
+      summary: List Instill Credit purchases.
+      description: |-
+        [Instill Credit](https://www.instill.tech/docs/vdp/credit) can be
+        purchased in order to extend the coverage of a given subscription tier.
+        This endpoint returns a paginated list of credit purchases.
+
+        On Instill Core, this endpoint will return a 404 Not Found status.
+      operationId: MgmtPublicService_ListCreditPurchases
+      responses:
+        "200":
+          description: A successful response.
+          schema:
+            $ref: '#/definitions/v1betaListCreditPurchasesResponse'
+        "401":
+          description: Returned when the client credentials are not valid.
+          schema: {}
+        default:
+          description: An unexpected error response.
+          schema:
+            $ref: '#/definitions/googlerpcStatus'
+      parameters:
+        - name: owner_name
+          description: |-
+            The user or organization to which the credit belongs.
+            Format: `{[users|organizations]}/{id}`.
+          in: path
+          required: true
+          type: string
+          pattern: '[^/]+/[^/]+'
+        - name: page_size
+          description: |-
+            The maximum number of purchase entries to return. If this parameter is
+            unspecified, at most 10 definitions will be returned. The cap value for
+            this parameter is 100 (i.e. any value above that will be coerced to 100).
+          in: query
+          required: false
+          type: integer
+          format: int32
+        - name: page
+          description: Page number.
+          in: query
+          required: false
+          type: integer
+          format: int32
```

Comment:
@orangecoloured @EiffelFly please notice this and provide feedback. We already implement this pagination payload for the component definition list.

While I think we should keep payloads consistent for the same pagination strategy (cursor-based vs offest & limit), I do not think we  should have a single payload for both use cases, which are quite different. When implementing pagination, it is good UX to be able to access a given page number, and we shouldn't burden the frontend loading all the pages and then building the page index.

I'd have one payload for cursor-based pagination (`next_page_token`) and one for offset-based pagination (`page_size` + `page`).

Reply:
I agree that cursor based is a better design on this display 🤝 And I also agree we don't need to maintain a singular pagination design at this point, but we might want to think about it when the API is ready to converge and we are marching to official launch